File: src/Neuralm.Presentation.Web/src/messages/responses/AuthenticateResponse.ts
import Response from './Response';

/**
 * Represents the AuthenticateResponse interface.
 */
export default class AuthenticateResponse extends Response {
  public accessToken: string;
  public userId: string;

  /**
   * Initializes a new instance of the AuthenticateResponse class.
   */
  constructor(id: string, requestId: string, dateTime: Date, message: string, success: boolean, userId: string, accessToken: string) {
    super(id, requestId, dateTime, message, success);
    this.userId = userId;
    this.accessToken = accessToken;
  }
}

File: src/Neuralm.Services/Neuralm.Services.UserService/Neuralm.Services.UserService.Messages/AuthenticateResponse.cs
using Neuralm.Services.Common.Messages.Abstractions;
using System;

namespace Neuralm.Services.UserService.Messages
{
    /// <summary>
    /// Represents the <see cref="AuthenticateResponse"/> class.
    /// </summary>
    public class AuthenticateResponse : Response
    {
        /// <summary>
        /// Gets the access token.
        /// </summary>
        public string AccessToken { get; set; }

        /// <summary>
        /// Gets the user id.
        /// </summary>
        public Guid UserId { get; set; }

        /// <summary>
        /// Initializes an instance of the <see cref="AuthenticateResponse"/> class.
        /// </summary>
        /// <param name="requestId">The request id.</param>
        /// <param name="userId">The user id.</param>
        /// <param name="accessToken">The access token.</param>
        /// <param name="message">The message.</param>
        /// <param name="success">The success flag.</param>
        public AuthenticateResponse(Guid requestId, Guid userId, string accessToken = "", string message = "", bool success = false) : base(requestId, message, success)
        {
            AccessToken = accessToken;
            UserId = userId;
        }
        
        /// <summary>
        /// Initializes a new instance of the <see cref="AuthenticateResponse"/> class.
        /// SERIALIZATION CONSTRUCTOR!
        /// </summary>
        public AuthenticateResponse()
        {
            
        }
    }
}
